### PR TITLE
Add streaming KZG

### PR DESCRIPTION
**Pull request by huyuncong**
## Description

Add streaming KZG used in Gemini.
I need help to fix the test for the poseidon:  https://github.com/arkworks-rs/poly-commit/blob/master/src/lib.rs#L1290

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (master)
- [ ] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Wrote unit tests
- [ ] Updated relevant documentation in the code
- [ ] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: src/streaming_kzg/mod.rs [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+//! # // XXX. if you change the following lines,
+//! # // please note that documentation below might break.
+//! # let f = vec![Fr::from(1u64), Fr::from(2u64), Fr::from(4u64), Fr::from(8u64)];
+//! # let commitment  = ck.commit(&f);
+//! # let alpha = Fr::from(42u64);
+//! # let (evaluation, proof) = ck.open(&f, &alpha);
+//! # use ark_poly_commit::streaming_kzg::VerifierKey;
+//! # let vk = VerifierKey::from(&ck);
+//! # assert!(vk.verify(&commitment, &alpha, &evaluation, &proof).is_ok())
+//! ````
```
Comment (Pratyush) [inline src/streaming_kzg/mod.rs:64]:
```suggestion
//! ```
```

File: src/streaming_kzg/mod.rs [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+) -> DensePolynomial<E::Fr> {
+    let mut res = DensePolynomial::from_coefficients_vec(vec![E::Fr::zero()]);
+    for (j, (_x_j, y_j)) in eval_points.iter().zip(evals.iter()).enumerate() {
+        let l_poly = lang[j].mul(sca_inverse[j] * y_j);
+        res = (&res).add(&l_poly);
+    }
+    res
+}
+
+/// The polynomial in \\(\FF\\) that vanishes in all the points `points`.
```
Comment (Pratyush) [inline src/streaming_kzg/mod.rs:286]: This takes quadratic time, no?

**Comment by Pratyush** [comment]
looks good to me modulo the quadratic time question.

File: src/streaming_kzg/mod.rs (hunk 2 above)
Comment (huyuncong) [reply src/streaming_kzg/mod.rs:285]: Yes. It is quadratic to the number of evaluation points. In Gemini or Marlin, there is only a constant number of evaluation points.

**Pull request merged**